crates/cards/Cargo.toml

```diff
@@ -8,7 +8,9 @@ edition = "2021"
 [dependencies]
 time = { version = "0.3.20", features = ["serde", "serde-well-known", "std"] }
 error-stack = "0.3.1"
+serde = { version = "1.0.155", features = ["derive"] }
+serde_json = "1.0.94"
```

Comment:
I don't see any usages of these crates. Do you plan to use them in this PR?

Reply:
@SanchithHegde  Yes, how do I add serialization and deserialization support for the new structs ?

Reply:
For serialization of tuple structs with a single field, you should be able to use [`#[serde(transparent)]`](https://serde.rs/container-attrs.html#transparent). As for deserialization, you'd have to [implement a custom deserializer](https://serde.rs/impl-deserialize.html) which internally uses the `TryFrom` implementations.

Reply:
@SanchithHegde can you help me with these : 

- To derive serializable implementation of `StrongSecret<u16>` or  `StrongSecret<u8>`, `u8` and `u16` primitive types should implement `SerializableSecret` trait defined in masking/serde.rs. I tried implementing it but   It seems I cannot implement it within cards and I need to add the impl in the crate where `SerializableSecret` trait it defined.

- does this deserialize look as expected:  
```rust
impl<'de> Deserialize<'de> for CardSecurityCode {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: serde::Deserializer<'de>,
    {
        let csc = u16::deserialize(deserializer)?;
        csc.try_into().map_err(de::Error::custom)
    }
}
```

Reply:
> To derive serializable implementation of `StrongSecret<u16>` or  `StrongSecret<u8>`, `u8` and `u16` primitive types should implement `SerializableSecret` trait defined in masking/serde.rs. I tried implementing it but   It seems I cannot implement it within cards and I need to add the impl in the crate where `SerializableSecret` trait it defined.

Yeah, you can implement the trait for `u8` and `u16` in the `masking` crate itself.

> does this deserialize look as expected:

Looks good, could you add tests to verify that the deserialization works as expected, and that invalid card security codes / expiration months / expiration years raise an error?